GenericDiffractometer.py

```diff
@@ -635,8 +693,41 @@ def manual_centring(self):
     def automatic_centring(self):
         raise NotImplementedError
 
+    def centring_motor_moved(self, pos):
+        if time.time() - self.centring_time > 1.0:
+            self.invalidate_centring()
+        self.emit_diffractometer_moved()
+
+    def invalidate_centring(self):
+        if self.current_centring_procedure is None and self.centring_status["valid"]:
+            self.centring_status={"valid":False}
+            self.emit_progress_message("")
+            self.emit('centringInvalid', ())
+
+    def emit_diffractometer_moved(self, *args):
+        self.emit("diffractometerMoved", ())
+
     def motor_positions_to_screen(self, centred_positions_dict):
-        raise NotImplementedError
+        if self.use_sample_centring:
```

Comment:
I would suggest also move this to your specific implementation. If in the end all sites apart from us use this then we can move it to GenericDiffractometer.